sklearn/preprocessing/label.py

```diff
@@ -125,6 +125,8 @@ def transform(self, y):
         y : array-like of shape [n_samples]
         """
         check_is_fitted(self, 'classes_')
+        if _num_samples(y) == 0:
+            return np.array([])
```

Comment:
I think it might be better to put it after `column_or_1d` (first do input validation), WDYT?

Reply:
hmm that seems correct